File: 351003/Zarenok/demo/publisher/src/main/java/com/example/demo/dto/requests/AuthorRequestTo.java
package com.example.demo.dto.requests;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.Data;

@Data
public class AuthorRequestTo {
    @NotBlank(message = "Login is required")
    @Size(min = 2, max = 64, message = "Login must be between 2 and 64 characters")
    @JsonProperty("login")
    private String login;

    @NotBlank(message = "Password is required")
    @Size(min = 8, max = 128, message = "Password must be between 2 and 128 characters")
    @JsonProperty("password")
    private String password;

    @NotBlank(message = "Firstname is required")
    @Size(min = 2, max = 64, message = "Firstname must be between 2 and 64 characters")
    @JsonProperty("firstname")
    private String firstname;

    @NotBlank(message = "Lastname is required")
    @Size(min = 2, max = 64, message = "Lastname must be between 2 and 64 characters")
    @JsonProperty("lastname")
    private String lastname;

    @JsonProperty("role")
    private String role;
}

File: 351001/shevchenkoAD/DistComp1/Application/DTOs/Requests/AuthorRequestTo.cs
using System.ComponentModel.DataAnnotations;
using Application.DTOs.Abstractions;

namespace Application.DTOs.Requests;

public record AuthorRequestTo(
    long Id,
    [Required]
    [StringLength(64, MinimumLength = 2)]
    string Login,
    [Required]
    [StringLength(128, MinimumLength = 8)]
    string Password,
    [Required]
    [StringLength(64, MinimumLength = 2)]
    string Firstname,
    [Required]
    [StringLength(64, MinimumLength = 2)]
    string Lastname
)
    : BaseRequestTo(Id);